ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/privacy/eea/PrivateEeaNonceProviderTest.java

```diff
@@ -103,8 +104,9 @@ public void moreThanOneMatchingLegacyGroupThrowsException() {
 
   @Test
   public void enclaveThrowingAnExceptionResultsInErrorResponse() {
-    when(enclave.findPrivacyGroup(any())).thenThrow(new EnclaveException("Enclave Failed"));
-    assertThatExceptionOfType(EnclaveException.class)
+    when(enclave.findPrivacyGroup(any()))
+        .thenThrow(new EnclaveServerException(500, "Enclave Failed"));
+    assertThatExceptionOfType(EnclaveClientException.class)
```

Comment:
Is this correct? Looks like we should assert that the thrown exception is an EnclaveServerException instead of the EnclaveClientException (or maybe I'm missing something).